beacon-chain/slasher/process_slashings.go

```diff
@@ -0,0 +1,69 @@
+package slasher
+
+import (
+	"context"
+
+	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/blocks"
+	"github.com/prysmaticlabs/prysm/shared/bytesutil"
+)
+
+// Verifies attester slashings, logs them, and submits them to the slashing operations pool
+// in the beacon node if they pass validation.
+func (s *Service) processAttesterSlashings(ctx context.Context, slashings []*ethpb.AttesterSlashing) {
+	for _, sl := range slashings {
+		if err := s.verifyAttSignature(ctx, sl.Attestation_1); err != nil {
+			log.WithField("a", sl.Attestation_1).Debug(
+				"Invalid signature for attestation in detected slashing offense",
+			)
+			continue
+		}
+		if err := s.verifyAttSignature(ctx, sl.Attestation_2); err != nil {
+			log.WithField("a", sl.Attestation_2).Debug(
+				"Invalid signature for attestation in detected slashing offense",
+			)
+			continue
+		}
+		// TODO(#8331): Log the slashing event.
+
+		// TODO(#8331) Submit to the slashing operations pool.
+	}
+}
+
+// Verifies proposer slashings, logs them, and submits them to the slashing operations pool
+// in the beacon node if they pass validation.
+func (s *Service) processProposerSlashings(ctx context.Context, slashings []*ethpb.ProposerSlashing) {
+	for _, sl := range slashings {
+		if err := s.verifyBlockSignature(ctx, sl.Header_1); err != nil {
+			log.WithField("a", sl.Header_1).Debug(
+				"Invalid signature for block header in detected slashing offense",
+			)
+			continue
+		}
+		if err := s.verifyBlockSignature(ctx, sl.Header_2); err != nil {
```

Comment:
Don't we have the data root of the block header already? If we get a lot of slashings on the network it might just be constant hashing.

Reply:
Whoops didnt mean to approve